AC-906 : Re-Designed Home Screen UI

sahuadarsh0 opened the pull request:
## Description of what I changed
Home Screen UI is re-designed both for dark mode and light mode

### New Changes I have made
1. Removed all bitmap and bind drawable resources instead used vector files for faster loading across all screen sizes. Also removed changeColorOfDashboardIcons cause icons look great in both dark and light mode
2. Changed position of switching to different session location to bottom navigation as suggested by @Akshatji800 
3. Shifted Settings from the action bar to bottom navigation
4. Added `About Us` as 4th option later to be replaced by Profile dashboard as suggested by @LuGO0 


## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it -->
<!--- Just add the issue number at the end: -->
JIRA Issue: https://issues.openmrs.org/browse/AC-906

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1). 

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

## ScreenShots
### OLD
![old](https://user-images.githubusercontent.com/25864174/113519097-bd1fff80-95a7-11eb-8ad9-e85dcdd48176.png)


### NEW
Device Mi A3 API 29

![New](https://user-images.githubusercontent.com/25864174/113519032-7af6be00-95a7-11eb-992b-8722e5e0a421.jpg)![New](https://user-images.githubusercontent.com/25864174/113519034-7cc08180-95a7-11eb-9d98-f1efe6dfb5c9.jpg)

LuGO0 commented:
@sahuadarsh0 please lookupto my reviews in the previous PR please dont delete PRs there was a lot of conversation in that PR which is quite useful !!

sahuadarsh0 commented:
@LuGO0 I have addressed all the reviews in this PR, and I will keep this in mind for next time. The problem was when new PRs merged into the base then this PR showed conflict in merging and the CI builds were failing checks even before starting. I tried to undo things but that didn't work out well. That's why I have mentioned the old PR link in comment

LuGO0 commented on openmrs-client/src/main/res/menu/bottom_nav_menu.xml:
```diff
@@ -0,0 +1,38 @@
+    tools:context="org.openmrs.mobile.activities.ACBaseActivity">
+
+
+    <item
```
reformat this file @sahuadarsh0

sahuadarsh0 replied:
ok done

LuGO0 commented on openmrs-client/src/main/java/org/openmrs/mobile/activities/dashboard/DashboardFragment.kt:
```diff
@@ -148,65 +124,6 @@ class DashboardFragment : ACBaseFragment<DashboardContract.Presenter>(), Dashboa
-    override fun onDestroy() {
-        super.onDestroy()
-        unbindDrawableResources()
-    }
```
how are we taking care of unbinding vector resources @sahuadarsh0 just curious want to know more about it !!

LuGO0 commented on openmrs-client/src/main/java/org/openmrs/mobile/activities/dashboard/DashboardFragment.kt:
```diff
@@ -13,51 +13,32 @@
     private var mActiveVisitsView: RelativeLayout? = null
     private var mCaptureVitalsView: RelativeLayout? = null
     private var mProviderManagementView: RelativeLayout? = null
-    private var mBitmapCache: SparseArray<Bitmap?>? = null
```
@sahuadarsh0 I think there is a memory leak in there since the fragment will go to backstack but will hold to these views which should have been nullified in onDestroyView ? since when the fragment is called back from the backStack onCreateView will be called again and these views will be initialized again?

LuGO0 replied:
I think we can make those variables local to the function scope rather than keeping them at activity scope since they are being used only inside a function and keeping just the binding variable on the activity scope.

sahuadarsh0 replied:
we are creating png file for each pixel densities of the 4 dashboard icons and then binding them and unbinding them on onDestroy() but to use the same icons on bottom navigations we must use vector drawable or one png file ... different density png files are resulting into **smaller icons** in the bottom navigation, that's why I used vector files. But I can't find another way of unbinding them from the dashboardFragment.

LuGO0 replied:
we will need some help here I dont have much information regarding this I will have to read a bit about it before suggesting anything 👍🏼

LuGO0 commented:
Hi @rishabh-997 waiting for your review on this. It works fine after build

LuGO0 commented:
Hi @sahuadarsh0 thanks for continuously resolving the merge conflicts, we have just delayed merging of all PRs in the backlog. I will let you know when we will be ready to merge this PR and then we can work on this PR again.

sahuadarsh0 commented:
@LuGO0 ok, sure thanks for the follow-up...
